src/components/sidebar/SafeList/index.tsx

```diff
@@ -59,13 +64,55 @@ export const _shouldExpandSafeList = ({
 const MAX_EXPANDED_SAFES = 3
 const NO_SAFE_MESSAGE = 'Create a new safe or add'
 
+export const addActionsToTxs = (
+  chainId: string,
+  address: string,
+  page: TransactionListPage,
+  isSafeOwned: boolean,
+  txsActionsByChain: Record<string, Record<string, SafeActions>>,
+  walletAddress?: string,
+) => {
+  if (page.results.length === 0) return txsActionsByChain
+
+  if (isSafeOwned) {
+    const txs = page.results.filter(isTransactionListItem)
+
+    txs.reduce((acc, tx) => {
+      if (isSignableBy(tx.transaction, walletAddress || '')) {
+        acc[chainId] ??= {}
+        acc[chainId][address] ??= {}
+        acc[chainId][address].signing = Number(acc[chainId]?.[address].signing || 0) + 1
+      }
+      return acc
+    }, txsActionsByChain)
+  }
+
+  txsActionsByChain[chainId] = {
+    ...(txsActionsByChain[chainId] ?? {}),
+    [address]: {
+      ...txsActionsByChain[chainId]?.[address],
+      queued: `${page.results.filter(isTransactionListItem).length}${page.next ? '+' : ''}`,
+    },
+  }
```

Comment:
I'd suggest assigning the properties directly here as well. [Spreading a large object is not as performant](https://www.youtube.com/watch?v=tcZbY-Q0TIE).